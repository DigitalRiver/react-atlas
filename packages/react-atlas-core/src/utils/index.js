import events from "./events";
import prefixer from "./prefixer";
// import time from './time';
import utils from "./utils";
import classNames from "./classNames";

export { events, prefixer, utils, classNames };
