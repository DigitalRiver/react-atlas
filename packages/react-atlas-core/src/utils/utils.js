export default {
  angleFromPositions(cx, cy, ex, ey) {
    const theta = Math.atan2(ey - cy, ex - cx) + Math.PI / 2;
    return theta * 180 / Math.PI;
  },
  angle360FromPositions(cx, cy, ex, ey) {
    const angle = this.angleFromPositions(cx, cy, ex, ey);
    return angle < 0 ? 360 + angle : angle;
  },
  range(start = 0, stop = null, step = 1) {
    let [_start, _stop] = [0, start];
    if (stop !== null) {
      [_start, _stop] = [start, stop];
    }
    const length = Math.max(Math.ceil((_stop - _start) / step), 0);
    const range = Array(length);

    for (let idx = 0; idx < length; idx++, _start += step) {
      range[idx] = _start;
    }

    return range;
  },
  round(number, decimals) {
    if (!isNaN(parseFloat(number)) && isFinite(number)) {
      const decimalPower = Math.pow(10, decimals);
      return Math.round(parseFloat(number) * decimalPower) / decimalPower;
    }
    return NaN;
  },
  getViewport() {
    return {
      "height": window.innerHeight || document.documentElement.offsetHeight,
      "width": window.innerWidth || document.documentElement.offsetWidth
    };
  },
  cloneObject(object) {
    return JSON.parse(JSON.stringify(object));
  },
  inputTypeForPrototype(prototype) {
    if (prototype === Date) {
      return "date";
    }
    if (prototype === Number) {
      return "number";
    }
    if (prototype === Boolean) {
      return "checkbox";
    }
    return "text";
  },
  prepareValueForInput(value, type) {
    if (type === "date") {
      return new Date(value).toISOString().slice(0, 10);
    }
    if (type === "checkbox") {
      return value ? "on" : null;
    }
    return value;
  },
  getSelection(elem) {
    let start, end, rangeElem, clone;

    if (elem.selectionStart !== undefined) {
      start = elem.selectionStart;
      end = elem.selectionEnd;
    }
    else {
      try {
        elem.focus();
        rangeElem = elem.createTextRange();
        clone = rangeElem.duplicate();

        rangeElem.moveToBookmark(document.selection.createRange().getBookmark());
        clone.setEndPoint('EndToStart', rangeElem);

        start = clone.text.length;
        end = start + rangeElem.text.length;
      }
      catch (e) { /* not focused or not visible */ }
    }

    return { start, end };
  },
  setSelection(elem, selection) {
    let rangeElem;

    try {
      if (elem.selectionStart !== undefined) {
        elem.focus();
        elem.setSelectionRange(selection.start, selection.end);
      }
      else {
        elem.focus();
        rangeElem = elem.createTextRange();
        rangeElem.collapse(true);
        rangeElem.moveStart('character', selection.start);
        rangeElem.moveEnd('character', selection.end - selection.start);
        rangeElem.select();
      }
    }
    catch (e) { /* not focused or not visible */ }
  }
};
