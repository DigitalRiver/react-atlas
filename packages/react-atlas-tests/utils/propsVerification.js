export function verifyPropsDefaultValue(component, expectedProps) {
  //Compares each property value of the component with the expected value

  var result = true;
  var partialResult;

  for (var [compPropName, compPropValue] of expectedProps.entries()) {
    partialResult = component.props()[compPropName] == compPropValue;

    if (!partialResult) {
      console.log(
        "[" +
          compPropName +
          "] -> Got: " +
          component.props()[compPropName] +
          " - expected: " +
          compPropValue
      );
      result = false;
    }
  }
  return result;
}
