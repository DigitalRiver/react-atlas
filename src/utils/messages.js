const messages = {
  "requiredMessage": "This field is required",
  "selectOne": "Select One",
  "onSubmitAction": "Pass either onSubmit or action",
  "missingRequired": "Required input value is empty",
  "expandAll": "Expand All"
};

export default messages;
