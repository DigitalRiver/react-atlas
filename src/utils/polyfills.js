import "core-js/fn/array/from";
import "core-js/fn/array/iterator";
import "core-js/fn/map";
import "core-js/fn/string/starts-with";
import "core-js/fn/symbol";
