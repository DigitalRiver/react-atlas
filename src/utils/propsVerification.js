export function verifyPropsDefaultValue(component, expectedProps) {
  //Compares each property value of the component with the expected value

  let result = true;
  let partialResult;

  for (let [compPropName, compPropValue] of expectedProps.entries()) {
    partialResult = component.props()[compPropName] === compPropValue;

    if (!partialResult) {
      console.log(
        "[" +
          compPropName +
          "] -> Got: " +
          component.props()[compPropName] +
          " - expected: " +
          compPropValue
      );
      result = false;
    }
  }
  return result;
}
