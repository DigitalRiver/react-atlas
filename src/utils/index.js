import events from "./events";
import prefixer from "./prefixer";
// import time from './time';
import utils from "./utils";

export { events, prefixer, utils };
