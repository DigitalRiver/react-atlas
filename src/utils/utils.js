import React from "react";
import messages from "./messages.js";

export default {
  angleFromPositions(cx, cy, ex, ey) {
    const theta = Math.atan2(ey - cy, ex - cx) + Math.PI / 2;
    return theta * 180 / Math.PI;
  },
  angle360FromPositions(cx, cy, ex, ey) {
    const angle = this.angleFromPositions(cx, cy, ex, ey);
    return angle < 0 ? 360 + angle : angle;
  },
  range(start = 0, stop = null, step = 1) {
    let [_start, _stop] = [0, start];
    if (stop !== null) {
      [_start, _stop] = [start, stop];
    }
    const length = Math.max(Math.ceil((_stop - _start) / step), 0);
    const range = Array(length);

    for (let idx = 0; idx < length; idx++, _start += step) {
      range[idx] = _start;
    }

    return range;
  },
  round(number, decimals) {
    if (!isNaN(parseFloat(number)) && isFinite(number)) {
      const decimalPower = Math.pow(10, decimals);
      return Math.round(parseFloat(number) * decimalPower) / decimalPower;
    }
    return NaN;
  },
  getViewport() {
    return {
      "height": window.innerHeight || document.documentElement.offsetHeight,
      "width": window.innerWidth || document.documentElement.offsetWidth
    };
  },
  cloneObject(object) {
    return JSON.parse(JSON.stringify(object));
  },
  inputTypeForPrototype(prototype) {
    if (prototype === Date) {
      return "date";
    }
    if (prototype === Number) {
      return "number";
    }
    if (prototype === Boolean) {
      return "checkbox";
    }
    return "text";
  },
  prepareValueForInput(value, type) {
    if (type === "date") {
      return new Date(value).toISOString().slice(0, 10);
    }
    if (type === "checkbox") {
      return value ? "on" : null;
    }
    return value;
  },
  getSelection(elem) {
    let start, end, rangeElem, clone;

    if (typeof elem.selectionStart !== "undefined") {
      start = elem.selectionStart;
      end = elem.selectionEnd;
    } else {
      try {
        elem.focus();
        rangeElem = elem.createTextRange();
        clone = rangeElem.duplicate();

        rangeElem.moveToBookmark(
          document.selection.createRange().getBookmark()
        );
        clone.setEndPoint("EndToStart", rangeElem);

        start = clone.text.length;
        end = start + rangeElem.text.length;
      } catch (e) {
        /* not focused or not visible */
      }
    }

    return { start, end };
  },
  setSelection(elem, selection) {
    let rangeElem;

    try {
      if (typeof elem.selectionStart !== "undefined") {
        elem.focus();
        elem.setSelectionRange(selection.start, selection.end);
      } else {
        elem.focus();
        rangeElem = elem.createTextRange();
        rangeElem.collapse(true);
        rangeElem.moveStart("character", selection.start);
        rangeElem.moveEnd("character", selection.end - selection.start);
        rangeElem.select();
      }
    } catch (e) {
      /* not focused or not visible */
    }
  },
  getComponentName(comp) {
    if (comp.type) {
      if (
        Object.keys(comp.type).length === 0 &&
        comp.type.constructor === Object
      ) {
        return null;
      }
      return comp.type.displayName || comp.type.name || comp.type || null;
    }
    return null;
  },
  filter(object, excludes) {
    return Object.keys(object)
      .filter(obj => !excludes[obj])
      .reduce((obj, key) => {
        obj[key] = object[key];
        return obj;
      }, {});
  },
  getRequiredText(reqText) {
    return <span styleName={"error_text"}>{reqText}</span>;
  },
  getErrorMessage(errorMessage) {
    return <div styleName={"error_message"}>{errorMessage}</div>;
  },
  validate(inputValue, validator, propStatus, propMessage, required, event) {
    let status = null;
    let message = null;
    /* Execute custom validator and change state and error messages accordingly */
    if (typeof validator === "function") {
      let validationObject = validator(event, inputValue);
      if (typeof validationObject !== "undefined") {
        if (
          typeof validationObject === "boolean" &&
          validationObject === true
        ) {
          status = propStatus;
          message = propMessage;
        } else if (
          typeof validationObject === "boolean" &&
          validationObject === false
        ) {
          status = "error";
        } else {
          status = validationObject.status;
          message = validationObject.message;
        }
      }
    } else if (
      (typeof inputValue === "undefined" || inputValue === "") &&
      (required || typeof required === "string" && required === "")
    ) {
      /* If the field is required, and it has no value, change state and display error message */
      message = messages.requiredMessage;
      status = "error";
    }
    return { "status": status, "message": message };
  }
};
