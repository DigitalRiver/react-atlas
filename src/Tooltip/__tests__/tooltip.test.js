import React from "react";
import { mount } from "enzyme";
import { Tooltip } from "../index";

import { verifyPropsDefaultValue } from "../../utils/propsVerification";

import renderer from "react-test-renderer";

describe("Test Tooltip component render", () => {
  it("Render correctly", () => {
    const tree = renderer
      .create(
        <Tooltip
          tooltip={"Hey I'm a tooltip"}
          position={"top"}
          icon={"<i class='fa fa-github'></i>"}
          tooltipDelay={3000}
          active={false}
          className={"class"}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Testing Tooltip component", () => {
  it("Tooltip component - Properties basic test", function() {
    const comp = mount(
      <Tooltip
        tooltip={"Hey I'm a tooltip"}
        position={"top"}
        icon={"<i class='fa fa-github'></i>"}
        tooltipDelay={3000}
        active={false}
        className={"class"}
      />
    );
    const expectedProps = new Map([
      ["tooltip", "Hey I'm a tooltip"],
      ["position", "top"],
      ["tooltipDelay", 3000],
      ["active", false],
      ["className", "class"]
    ]);

    expect(verifyPropsDefaultValue(comp, expectedProps)).toEqual(true);
  });

  it("Tooltip component - Mouse in with delays", function() {
    const comp = mount(<Tooltip tooltip={"Hey I'm a tooltip"} delay={1000} />);
    jest.useFakeTimers();
    expect(comp.state().active).toEqual(false);
    comp.simulate("mouseEnter");
    jest.runAllTimers();
    expect(comp.state().active).toEqual(true);
  });

  it("Tooltip component - Mouse in without delays", function() {
    const comp = mount(<Tooltip tooltip={"Hey I'm a tooltip"} />);
    jest.useFakeTimers();
    expect(comp.state().active).toEqual(false);
    comp.simulate("mouseEnter");
    jest.runAllTimers();
    expect(comp.state().active).toEqual(true);
  });

  it("Tooltip component - Mouse in/out", function() {
    const comp = mount(<Tooltip tooltip={"Hey I'm a tooltip"} />);

    comp.simulate("mouseEnter");
    expect(comp.state().active).toEqual(true);

    comp.simulate("mouseLeave");
    expect(comp.state().active).toEqual(false);
  });
});
