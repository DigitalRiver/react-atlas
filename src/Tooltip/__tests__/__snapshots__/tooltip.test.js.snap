// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tooltip component render Render correctly 1`] = `
<div
  className="class block tooltip tooltip-top"
  data-tooltip=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="tooltip"
  style={undefined}
>
  <i
    className="<i class='fa fa-github'></i>"
  />
  
</div>
`;
